### PR TITLE
Add instructions for splitting a github repo

### PR DESCRIPTION
**Pull request by rpdelaney**

### CODE REVIEW

**Comment by rpdelaney** [comment]
@jsclarridge pointed out that we have a documented work flow for this here: https://github.com/trussworks/Engineering-Playbook/blob/master/developing/vcs/git-repos.md#splitting-out-code-to-a-new-repository

I think git filter-branch is practically self-deprecated though as the git manpage strongly discourages its use. (See `git help filter-branch). I'll rework this PR to replace the vcs/git-repos section and link to it from the infrasec docs.

File: developing/vcs/git-repos.md [hunk 1]
```diff
@@ -109,3 +115,5 @@ git push origin .
 
 * [GitHub's documentation](https://help.github.com/en/github/using-git/splitting-a-subfolder-out-into-a-new-repository)
 * [Chris Gilmer’s full instructions](https://github.com/chrisgilmerproj/silliness#how-to-break-out-projects)
```
[the remark this answers is not in the document]
Comment (rpdelaney) [reply developing/vcs/git-repos.md]: 95d91d46293fcc17c22a7bfe59e11b3a0c21da35

**Comment by rpdelaney** [comment]
Originally I wanted to remove the origin first as an extra safety measure against accidentally pushing the filtered work tree. But `git-filter-branch` removes the origin remote itself after doing the filtering. With that in mind, rather than use `--force` anywhere I think it's better to skip the step where we remove the origin remote. 4399795

File: developing/vcs/git-repos.md [hunk 2]
```diff
@@ -65,47 +65,40 @@ If you need to split out code from one of your repositories into its own
...
 ```
 
-Filter out commits that change the specified directory:
+The `origin` remote will be removed in this process. Add it back with the new
+repo URL, and push the filtered work tree:
 
 ```sh
-git filter-branch --prune-empty --subdirectory-filter DIRNAME main
+git remote add origin git@github.com:OWNER/NEWREPO.git
+git push --set-upstream origin <main>
```
Comment (jsclarridge) [inline developing/vcs/git-repos.md:90]: Minor nitpick: this refers to the "new repo URL". Move the steps about creating the new repo to before these instructions?

Comment (rpdelaney) [reply developing/vcs/git-repos.md:90]: I don't know what to make of line 93. I think anyone with the user story "As an engineer, I want to move files from one repository into another while preserving work tree history" knows how to create a repository on github. And if they don't, this line doesn't help. Why not just delete it?

Comment (jsclarridge) [reply developing/vcs/git-repos.md:90]: Its inclusion in the current version of the doc makes the sequence of steps clear. And retaining it helps to set up the discussion about the repo license and importing the repo into Terraform. Maybe the other PR reviewers have thoughts.

**Pull request merged**